Projects/Chat/Sources/Views/ChatScreenViewModel.swift

```diff
@@ -0,0 +1,227 @@
+import Combine
+import Flow
+import Kingfisher
+import Presentation
+import SwiftUI
+import hCore
+
+class ChatScreenViewModel: ObservableObject {
+    @Published var messages: [Message] = []
+    @Published var lastDeliveredMessage: Message?
+    @Published var isFetchingNext = false
+    @Published var scrollToMessage: Message?
+    @Published var chatInputVm: ChatInputViewModel = .init()
+    @Inject private var fetchMessagesClient: FetchMessagesClient
+    @Inject private var sendMessageClient: SendMessageClient
+    private var addedMessagesIds: [String] = []
+    private var nextUntil: String?
+    private var hasNext: Bool?
+    private var isFetching = false
+
+    init() {
+        chatInputVm.sendMessage = { [weak self] message in
+            Task { [weak self] in
+                await self?.send(message: message)
+            }
+        }
+        Timer.scheduledTimer(withTimeInterval: 5, repeats: true) { [weak self] timer in
+            guard let self = self else {
+                timer.invalidate()
+                return
+            }
+            Task { [weak self] in
+                await self?.fetch()
+            }
+        }
+        Task { [weak self] in
+            await self?.fetch()
+        }
+        let fileUploadManager = FileUploadManager()
+        fileUploadManager.resetuploadFilesPath()
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
+            AskForRating().ask()
```

Comment:
Is this one ever triggered? Does not seem like it. Maybe we can either we can remove this or make it work?

Reply:
It is working, on third time if you check implementation
```
 func ask() {
        guard !UserDefaults.standard.bool(forKey: userDefaultsCompletedKey) else { return }

        let numberOfSessions = UserDefaults.standard.value(forKey: "AskForRating") as? Int ?? 0

        if numberOfSessions >= 3 {
            UserDefaults.standard.set(true, forKey: userDefaultsCompletedKey)
            if let scene = UIApplication.shared.currentScene {
                SKStoreReviewController.requestReview(in: scene)
            }
        }
    }
```